sdk/database/helper/connutil/sql_test.go

```diff
@@ -0,0 +1,56 @@
+package connutil
+
+import (
+	"context"
+	"net/url"
+	"testing"
+)
+
+func TestSQLPasswordChars(t *testing.T) {
+	testCases := []struct {
+		Username string
+		Password string
+	}{
+		{"postgres", "password{0}"},
+		{"postgres", "pass:word"},
+		{"postgres", "pass/word"},
+		{"postgres", "p@ssword"},
+		{"postgres", "pass\"word\""},
+		// Much to my surprise, CREATE USER "{{password}}" PASSWORD 'foo' worked.
```

Comment:
PG treats anything quoted as a literal and literals can be used for roles and database names.